### DISCUSSION TITLE
More details on how versioning works

### DISCUSSION THREAD

**Discussion by cesss** [microsoft/vcpkg#19049] [Q&A]
I have read the versioning documentation, but, however, I wasn't able to find the details on how several versions of the same package simultaneously live in the same tree. I did find that they seem to be in a build tree that has hashes, but, two questions arise:

1. Which is the real layout of that tree?
2. How do actually packages get access to a specific version? I mean, if a package doesn't use cmake, does it get access to the desired version via CPPFLAGS and LDFLAGS? How? (I know how to ask for a specific version, what I don't know is the mechanism which lets packages access the include and library directories of the version it requested).

**Comment by autoantwort** [comment]
> I wasn't able to find the details on how several versions of the same package simultaneously live in the same tree

Afaik they don't. There is only one version of each lib at the same time.  
To 2: There is only one version, so you don't have this problem

**Comment by cesss** [comment]
> > I wasn't able to find the details on how several versions of the same package simultaneously live in the same tree
> 
> Afaik they don't. There is only one version of each lib at the same time.
> To 2: There is only one version, so you don't have this problem

The examples in the versioning documentation show different packages getting different versions of their dependencies (because of different minimum version requirements), so I assumed different versions of the same package can coexist. Isn't this the case?

**Comment by autoantwort** [comment]
> The examples in the versioning documentation show different packages getting different versions of their dependencies 

Where do I found this in the examples? 

> Isn't this the case?

Afaik no

**Comment by dg0yt** [comment]
The tree of different versions is the git history, not the filesystem of a particular checkout.

**Comment by PhoebeHui** [comment]
@cesss, does this still be a question for you?
